lib/membrane/rtp/packet.ex

```diff
@@ -100,17 +126,38 @@ defmodule Membrane.RTP.Packet do
   def parse(_binary, _parse_payload?), do: {:error, :malformed_packet}
 
   defp parse_header_extension(binary, header_present?)
-  defp parse_header_extension(binary, false), do: {:ok, {nil, binary}}
+  defp parse_header_extension(binary, false), do: {:ok, {[], binary}}
 
   defp parse_header_extension(
-         <<profile_specific::binary-size(2), data_len::16, data::binary-size(data_len)-unit(32),
+         <<_profile_specific::binary-size(2), data_len::16, data::binary-size(data_len)-unit(32),
            rest::binary>>,
          true
        ) do
-    extension = %Header.Extension{profile_specific: profile_specific, data: data}
-
-    {:ok, {extension, rest}}
+    extensions = parse_data_while_possible(data, [])
+    {:ok, {extensions, rest}}
   end
 
   defp parse_header_extension(_binary, true), do: :error
+
+  defp parse_data_while_possible(<<>>, acc), do: acc
+
+  defp parse_data_while_possible(data, acc) do
+    case parse_one_byte_header(data) do
+      {:ok, {data, rest}} -> parse_data_while_possible(rest, [data | acc])
+      {:ok, rest} -> parse_data_while_possible(rest, acc)
+    end
+  end
+
+  defp parse_one_byte_header(<<extension_header::bits-size(8), extensions::binary>>) do
+    <<local_identifier::integer-size(4), len_data::integer-size(4)>> = extension_header
+
+    if local_identifier == 0 do
+      {:ok, extensions}
+    else
+      len_data = len_data + 1
+      <<data::binary-size(len_data), next_extensions::binary>> = extensions
+      extension = %Header.Extension{profile_specific: local_identifier, data: data}
```

Comment:
`Header.Extension has now wrong typespec 👀 
```elixir
@type t :: %__MODULE__{
          profile_specific: binary(),
          data: binary()
        }
```
`profile_specific` in RFC 8285 are used to distinguish between one- and two-byte headers. As we target one-byte headers for now, I think we should consider replacing it with `identifier` field.